.circleci/config.yml

```diff
@@ -200,6 +206,39 @@ jobs:
           paths:
             - ~/.m2
           key: v1-maven-{{ .Branch }}-{{ .BuildNum }}
+  server-java:
+    docker:
+      - image: circleci/openjdk:8-jdk
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            # when project files change, use increasingly general patterns to restore cache.
+            # vN prefix in case we ever need to regenerate all caches
+            - v1-maven-{{ .Branch }}-{{ .BuildNum }}
+            - v1-maven-{{ .Branch }}-
+            - v1-maven-
+      - run:
+          name: Build
+          command: |
+            cd server/java
+            ./scripts/clean.sh
+            ./scripts/build.sh
+      - run:
+          name: Tests
+          command: |
+            source ./.circleci/set_env_variables.sh
+            cd server/java
+            ./scripts/test.sh
+      - store_test_results:
+          path: server/java/target/surefire-reports
+      - codecov/upload:
+          file: server/java/target/site/jacoco/jacoco.xml
+          flags: server_java
```

Comment:
you need to add these flags to `.codecov.yml`, if you want to enforce checks on code coverage.